RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterQuickFixHelper.cs

```diff
@@ -0,0 +1,195 @@
+            {
+                _module.InsertLines(++lineNumber, line);
+            }
+            return lineNumber;
+        }
+        /// <summary>
+        /// Replaces a string/line in a Code Module and returns the line number
+        /// of the replaced line.
+        /// </summary>
+        public int ReplaceModuleLine(int line, string newContent)
+        {
+            _module.ReplaceLine(line, newContent);
+            return line;
+        }
+        public string ReplaceStringAtIndex(int lineNumber, string toReplace, string replacement, int startIndex)
+        {
+            var lineContent = GetModuleLine(lineNumber);
+            return ReplaceStringAtIndex(lineContent, toReplace, replacement, startIndex);
+        }
+        /// <summary>
+        /// Replaces a string at a specified zero-based index location in the original string.
+        /// </summary>
+        public string ReplaceStringAtIndex(string original, string toReplace, string replacement, int startIndex)
+        {
+            Debug.Assert(startIndex >= 0);
+            Debug.Assert(original.Contains(toReplace));
+
+            int stopIndex = startIndex + toReplace.Length - 1;
+            var prefix = original.Substring(0, startIndex);
+            var suffix = (stopIndex >= original.Length) ? string.Empty : original.Substring(stopIndex + 1);
+            var toBeReplaced = original.Substring(startIndex, stopIndex - startIndex + 1);
+
+            Debug.Assert(toBeReplaced.IndexOf(toReplace) == 0);
+            return prefix + toBeReplaced.Replace(toReplace, replacement) + suffix;
+        }
+        public bool VariableNameIsAlreadyUsed(string variableName)
+        {
+            if(_namesInUse.Count() == 0)
+            {
+                _namesInUse = GetVariableNamesUsedInProcedureContext();
+            }
+            return _namesInUse.Any(name => variableName.ToUpper().Equals(name.ToUpper()));
+        }
+
+        public IReadOnlyList<VBAParser.BlockStmtContext> GetBlockStmtContextsForContext(RuleContext context)
+        {
+            if (context is VBAParser.SubStmtContext)
+            {
+                return ((VBAParser.SubStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.FunctionStmtContext)
+            {
+                return ((VBAParser.FunctionStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.PropertyLetStmtContext)
+            {
+                return ((VBAParser.PropertyLetStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.PropertyGetStmtContext)
+            {
+                return ((VBAParser.PropertyGetStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.PropertySetStmtContext)
+            {
+                return ((VBAParser.PropertySetStmtContext)context).block().blockStmt();
+            }
+            return null;
```

Comment:
That would be because of the `IReadOnlyList` I think. `Enumerable.Empty<VBAParser.BlockStmtContext>().ToArray()` would do it.